testsuite/python/tests_common.py

```diff
@@ -596,6 +596,20 @@ def gay_berne_potential(r_ij, u_i, u_j, epsilon_0, sigma_0, mu, nu, k_1, k_2):
     return 4. * epsilon * (rr**-12 - rr**-6)
 
 
+def calc_derivative(func, x, eps=1.0e-7, index=None):
```

Comment:
could you please use [```scipy.misc.derivative```](https://docs.scipy.org/doc/scipy/reference/generated/scipy.misc.derivative.html). If that does not suit your needs you should definitely add a docstring to this function. Actually we could even use [```sympy```](https://www.sympy.org/en/index.html). But we probably don't want to depend on that.

Reply:
I will add a docstring. [scipy.misc.derivative](https://docs.scipy.org/doc/scipy/reference/generated/scipy.misc.derivative.html) does not allow for a multidimensional input.

Reply:
https://docs.scipy.org/doc/numpy/reference/generated/numpy.vectorize.html

Reply:
Does that help? I need to manipulate the element at a specific index while the rest stays the same.

Reply:
I added a docstring. I could have thrown an error if someone gives a multidimensional input and index=None, but I think it would be a "overkill". In the docstring it is stated, that the index needs to be specified if x is an array.